spec/lib/os_places_api/client_spec.rb

```diff
@@ -197,6 +197,27 @@
       end
     end
 
+    context "the postcode doesn't exist in the DPA dataset" do
+      before :each do
+        Postcode.where(postcode: postcode).map(&:destroy)
+      end
+
+      it "should query OS Places API and return results from LPI dataset" do
+        os_places_api_results[0]["LPI"] = os_places_api_results[0]["DPA"]
+        os_places_api_results[0].delete("DPA")
```

Comment:
I don't think it's safe to reuse the DPA result, as the LPI result is in a different format.

Here's a slimmed down extract from https://api.os.uk/search/places/v1/postcode?key=API_KEY_GOES_HERE&postcode=E18QS&dataset=DPA,LPI&output_srs=WGS84:

```json
{
    "results": [
        {
            "DPA": {
                "UPRN": "6714279",
                "ADDRESS": "1, WHITECHAPEL HIGH STREET, LONDON, E1 8QS",
                "POSTCODE": "E1 8QS",
                "LNG": -0.0729707,
                "LAT": 51.5144426,
                ...
            }
        },
        {
            "LPI": {
                "UPRN": "6714279",
                "ADDRESS": "1, WHITECHAPEL HIGH STREET, LONDON, TOWER HAMLETS, E1 8QS",
                "LNG": -0.0729707,
                "LAT": 51.5144426,
                ...
            }
        }
    ]
}
```

The LPI result has different keys to the DPA one, but the only one that is a concern is the lack of `POSTCODE` key, which could be problematic [here](https://github.com/alphagov/locations-api/blob/eb60df6cdd397bf97e8f83ba5717ec29063b510a/lib/os_places_api/location_builder.rb#L8).

Could you add an LPI result to the `os_places_api_results` variable at the top of the file (even though it has a lot of duplication to the DPA result)?